protocol-units/bridge/move-modules/sources/MOVE.move

```diff
@@ -0,0 +1,350 @@
+    fun init_module(resource_account: &signer) {
+        // Create the stablecoin with primary store support.
+        let constructor_ref = &object::create_named_object(resource_account, ASSET_SYMBOL);
+        primary_fungible_store::create_primary_store_enabled_fungible_asset(
+            constructor_ref,
+            option::none(),
+            utf8(ASSET_SYMBOL), /* name */
+            utf8(ASSET_SYMBOL), /* symbol */
+            8, /* decimals */
+            utf8(b"http://example.com/favicon.ico"), /* icon */
+            utf8(b"http://example.com"), /* project */
+        );
+
+        // Set ALL stores for the fungible asset to untransferable.
+        fungible_asset::set_untransferable(constructor_ref);
+
+        // All resources created will be kept in the asset metadata object.
+        let metadata_object_signer = &object::generate_signer(constructor_ref);
+
+        let minters = vector::empty<address>();
+        vector::push_back(&mut minters, @resource_addr);
+        vector::push_back(&mut minters, @origin_addr);
+
+        move_to(metadata_object_signer, Roles {
+            master_minter: @master_minter,
+            admin: signer::address_of(resource_account),
+            minters,
+            pauser: @pauser,
+            denylister: @denylister,
+        });
+
+        // Create mint/burn/transfer refs to allow creator to manage the stablecoin.
+        move_to(metadata_object_signer, Management {
+            extend_ref: object::generate_extend_ref(constructor_ref),
+            mint_ref: fungible_asset::generate_mint_ref(constructor_ref),
+            burn_ref: fungible_asset::generate_burn_ref(constructor_ref),
+            transfer_ref: fungible_asset::generate_transfer_ref(constructor_ref),
+        });
+
+        move_to(metadata_object_signer, State {
+            paused: false,
+        });
+
+        // Override the deposit and withdraw functions which mean overriding transfer.
+        // This ensures all transfer will call withdraw and deposit functions in this module and perform the necessary
+        // checks.
+        let deposit = function_info::new_function_info(
+            resource_account,
+            string::utf8(b"move_fa"),
+            string::utf8(b"deposit"),
+        );
+        let withdraw = function_info::new_function_info(
+            resource_account,
+            string::utf8(b"move_fa"),
+            string::utf8(b"withdraw"),
+        );
+        dispatchable_fungible_asset::register_dispatch_functions(
+            constructor_ref,
+            option::some(withdraw),
+            option::some(deposit),
+            option::none(),
+        );
+    }
+
+    /// Allow a spender to transfer tokens from the owner's account given their signed approval.
+    /// Caller needs to provide the from account's scheme and public key which can be gotten via the Aptos SDK.
+    public fun transfer_from(
```

Comment:
Might as well get rid of `transfer_from` once and for all?

Reply:
Removed `transfer_from` and re-ran the tests, they do run on CI too.